rust/ffi/node/src/table.rs

```diff
@@ -12,82 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::sync::mpsc;
-use std::thread;
 use arrow_array::RecordBatchIterator;
 use lance::dataset::{WriteMode, WriteParams};
 use lance::io::object_store::ObjectStoreParams;
+use std::sync::Arc;
 
-use neon::{prelude::*, types::Deferred};
-use neon::types::buffer::TypedArray;
 use crate::arrow::arrow_buffer_to_record_batch;
+use neon::prelude::*;
+use neon::types::buffer::TypedArray;
+use vectordb::Table;
 
-use crate::error::{Error, Result, ResultExt};
-use crate::{get_aws_creds, JsDatabase, runtime};
-
-type TableCallback = Box<dyn FnOnce(&mut vectordb::Table, &Channel, Deferred) + Send>;
+use crate::error::ResultExt;
+use crate::{get_aws_creds, runtime, JsDatabase};
 
-// Wraps a LanceDB table into a channel, allowing concurrent access
 pub(crate) struct JsTable {
-    tx: mpsc::Sender<JsTableMessage>,
+    pub table: Arc<Table>,
```

Comment:
As suggested in my other comment, if you don't use an `Arc` here it should be easier to clone and get an owned reference that can be mutated in the Rust API.

```suggestion
    pub table: Table,
```